cms/datasets/views.py

```diff
@@ -206,41 +237,57 @@ def get_objects(self, pks: list[Any]) -> QuerySet[Dataset]:  # pylint: disable=t
 
             # Use title and description from the API, the rest was grabbed from the compound ID
+            version_int = int(version)
             api_data_for_datasets.append(
                 {
                     "id": dataset_id,
                     "title": item_from_api.title,
                     "description": item_from_api.description,
                     "edition": edition,
-                    "version": version,
+                    "version": version_int,
                 }
             )
-            lookup_criteria.append((dataset_id, edition, version))
+            lookup_criteria.append((dataset_id, edition, version_int))
 
         existing_query = Q()
-        for namespace, edition, version in lookup_criteria:
-            existing_query |= Q(namespace=namespace, edition=edition, version=int(version))
+        for namespace, edition, ver in lookup_criteria:
+            existing_query |= Q(namespace=namespace, edition=edition, version=ver)
 
         existing_datasets_map: dict[tuple[str, str, int], Dataset] = {
             (ds.namespace, ds.edition, ds.version): ds for ds in Dataset.objects.filter(existing_query)
         }
 
-        datasets_to_create_instances: list[Dataset] = []
+        datasets_to_create: list[Dataset] = []
+        datasets_to_update: list[Dataset] = []
         for data in api_data_for_datasets:
-            data_version = int(data["version"])
-            if (data["id"], data["edition"], data_version) not in existing_datasets_map:
-                datasets_to_create_instances.append(
+            key = (data["id"], data["edition"], data["version"])
+            existing_dataset = existing_datasets_map.get(key)
+
+            if existing_dataset is None:
+                # Dataset doesn't exist, create it
+                datasets_to_create.append(
                     Dataset(
                         namespace=data["id"],
                         edition=data["edition"],
-                        version=data_version,
+                        version=data["version"],
                         title=data["title"],
                         description=data["description"],
                     )
                 )
+            else:
+                # Dataset exists, check if metadata needs updating
+                updated_fields = _update_dataset_metadata(
+                    existing_dataset,
+                    title=data["title"],
+                    description=data["description"],
+                )
+                if updated_fields:
+                    datasets_to_update.append(existing_dataset)
 
-        if datasets_to_create_instances:
-            Dataset.objects.bulk_create(datasets_to_create_instances)
+        if datasets_to_create:
+            Dataset.objects.bulk_create(datasets_to_create)
+        if datasets_to_update:
+            Dataset.objects.bulk_update(datasets_to_update, ["title", "description"])
```

Comment:
How about to mimic what you did on line 214 (`dataset.save(update_fields=updated_fields)`) we create a set called `all_updated_fields`, after line 285 we do `all_updated_fields.update(updated_fields)` and then finally here we use `all_updated_fields`? I know it's only 2 fields at the moment but might be good for future proofing and consistency.